packages/core/src/auth/utils.ts

```diff
@@ -634,28 +642,63 @@ export class ExtensionUse {
         }
 
         // Update state, so next time it is not first use
-        state.update(this.isExtensionFirstUseKey, false).then(undefined, e => {
-            getLogger().error('Memento.update failed: %s', (e as Error).message)
-        })
+        this.updateMemento(state, this.isExtensionFirstUseKey, false)
 
         return this.isFirstUseCurrentSession
     }
 
+    /**
+     * Checks if the extension has been updated since the last time it was used.
+     * If this is the first used version of the extension, it returns true.
+     * If the extension has been downgraded, it returns true.
+     *
+     * Caveat: This may return true even if an update hadn't occurred IF
+     * this function hasn't been called.
+     */
+    wasUpdated(state: vscode.Memento = globals.context.globalState) {
```

Comment:
Since we'll know the specific version that this item is released, maybe we can use that to our advantage to get a more accurate result? I'm not sure exactly how we intend to use this so maybe my recommendation is not useful.

Reply:
I think you are asking about checking updating from the regular extension to split versions. This check is just to see if the extension was updated since the last load in general, it is not specific to the split.